### PR TITLE
AWS - default rds port

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/aws/rds/rdsDefaultPort.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+            for (var instance of describeDBInstances.data) {
+                if (!instance.DBInstanceArn) continue;
+                if (!instance.Engine ||
+                    !(instance.Endpoint && instance.Endpoint.Port)) continue;
+                var defaultPort = defaultPorts.filter((d) => {
+                    return d.engine == instance.Engine && d.port == instance.Endpoint.Port;
+                });
+                if (defaultPort && defaultPort.length) {
+                    helpers.addResult(results, 2, 'RDS instance has default port',
```
Comment (mehakseedat63) [inline plugins/aws/rds/rdsDefaultPort.js:61]:
```suggestion
                    helpers.addResult(results, 2, 'RDS instance is running on default port',
```

File: plugins/aws/rds/rdsDefaultPort.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                        region, instance.DBInstanceArn);
+                } else {
+                    helpers.addResult(results, 0, 'RDS instance does not have default port',
```
Comment (mehakseedat63) [inline plugins/aws/rds/rdsDefaultPort.js:64]:
```suggestion
                    helpers.addResult(results, 0, 'RDS instance is not running on default port',
```

File: plugins/aws/rds/rdsDefaultPort.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+            if (!describeDBInstances.data.length) {
+                helpers.addResult(results, 0, 'No RDS instances found', region);
+                return rcb();
+            }
+
+            for (var instance of describeDBInstances.data) {
+                if (!instance.DBInstanceArn) continue;
+                if (!instance.Engine ||
+                    !(instance.Endpoint && instance.Endpoint.Port)) continue;
```
Comment (mehakseedat63) [inline plugins/aws/rds/rdsDefaultPort.js:56]: Why can't we make it one check?

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/aws/rds/rdsDefaultPort.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'RDS Default Port',
+    category: 'RDS',
+    domain: 'Databases',
+    description: 'Ensure RDS database instances are not using the default ports.',
+    more_info: 'Running RDS instances on default ports represent a potential security concern. Moving RDS instances ports to non-default ports will add an extra layer of security, protecting publicly accessible AWS RDS databases from brute force and dictionary attacks.',
```
Comment (mehakseedat63) [inline plugins/aws/rds/rdsDefaultPort.js:9]:
```suggestion
    more_info: 'Running RDS instances on default ports represent a potential security concern. Moving RDS instances ports to non-default ports will add an extra layer of security, protecting publicly accessible AWS RDS databases from brute force and dictionary attacks.',
```


rephrase this

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/aws/rds/rdsDefaultPort.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var defaultPorts = [
+            { 'engine': 'aurora-mysql', 'port': 3306 },
+            { 'engine': 'aurora-postgresql', 'port': 5432 },
+            { 'engine': 'custom-oracle-ee', 'port': 1521 },
+            { 'engine': 'mariadb', 'port': 3306 },
+            { 'engine': 'mysql', 'port': 3306 },
+            { 'engine': 'oracle-ee', 'port': 1521 },
```
Comment (AkhtarAmir) [inline plugins/aws/rds/rdsDefaultPort.js:26]: Why do we need redundancy? We can have just 1 value for oracle databases. Same goes for sqlserver

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**